kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1038 @@
+                lhs_contribution(internal_index, slave_index) = 0.0;
+            }
+        }
+        for (auto &slave_index : local_slave_index_vector)
+        {
+            rhs_contribution(slave_index) = 0.0;
+        }
+
+        rLHSContribution = lhs_contribution;
+        rRHSContribution = rhs_contribution;
+
+        KRATOS_CATCH("Applying Multipoint constraints failed ..");
+    }
+
+    /**
+     * @brief This method reconstructs the slave solution after Solving. 
+     * @param rModelPart Reference to the ModelPart containing the problem.
+     * @param A System matrix
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual)
+     */
+    void ReconstructSlaveSolutionAfterSolve(
+        ModelPart &rModelPart,
+        TSystemMatrixType &rA,
+        TSystemVectorType &rDx,
+        TSystemVectorType &rb)
+    {
+
+        const int number_of_constraints = static_cast<int>(mGlobalMasterSlaveRelations.size());
+        // Getting the beginning iterator
+
+        GlobalMasterSlaveRelationContainerType::iterator constraints_begin = mGlobalMasterSlaveRelations.begin();
+        ProcessInfo &r_current_process_info = rModelPart.GetProcessInfo();
+        //contributions to the system
+        VectorType master_weights_vector;
+        double constant = 0.0;
+        double slave_dx_value = 0.0;
+        IndexType slave_equation_id = 0;
+        EquationIdVectorType master_equation_ids = EquationIdVectorType(0);
+
+        for (int i_constraints = 0; i_constraints < number_of_constraints; i_constraints++)
+        {
+            GlobalMasterSlaveRelationContainerType::iterator it = constraints_begin + i_constraints;
+
+            slave_dx_value = 0.0;
+            //get the equation Ids of the constraint
+            (*it).EquationIdVector(slave_equation_id, master_equation_ids, r_current_process_info);
+            //calculate constraint's T and b matrices
+            (*it).CalculateLocalSystem(master_weights_vector, constant, r_current_process_info);
+            int master_index = 0;
+            for (auto &master_equation_id : master_equation_ids)
+            {
+                slave_dx_value += TSparseSpace::GetValue(rDx, master_equation_id) * master_weights_vector(master_index);
+                master_index++;
+            }
+            slave_dx_value += constant;
+
+            rDx[slave_equation_id] = slave_dx_value;
+        }
+    }
+
+    /**
+     * @brief this method checks if any of the nodes of the given rGeometry is marked SLAVE.
+     * @param rGeometry The geometry to check for.
+     */
+    bool HasSlaveNode(GeometryType& rGeometry)
+    {
+        bool slave_found = false;
+        const SizeType number_of_nodes = rGeometry.PointsNumber();
+        for (IndexType j = 0; j < number_of_nodes; j++)
+        {
+            if (rGeometry[j].IsDefined(SLAVE))
+                slave_found = rGeometry[j].Is(SLAVE);
+            if (slave_found)
```

Comment:
if you return true here (and false at the end of the function) then you don't need the var slave_found